app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    ext_list = []
+    for pers_dict in people:
+        person = Person(pers_dict["name"], pers_dict["age"])
+        ext_list.append(person)
+
+    for pers_dict in people:
+        person = Person.people[pers_dict["name"]]
+        if "wife" in pers_dict and pers_dict["wife"] is not None:
+            person.wife = Person.people.get(pers_dict["wife"])
+        elif "husband" in pers_dict and pers_dict["husband"] is not None:
+            person.husband = Person.people.get(pers_dict["husband"])
```

Comment:
The code assumes that the 'wife' or 'husband' key will always be present in the dictionary. It might be safer to use `pers_dict.get('wife')` and `pers_dict.get('husband')` to avoid potential `KeyError` exceptions if these keys are missing.